Team9Project/src/com/indragie/cmput301as1/DestinationAddActivity.java

```diff
@@ -1,33 +1,143 @@
+/* 
+ * Copyright (C) 2015 Andrew Zhong
+ * 
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 package com.indragie.cmput301as1;
 
-import android.app.Activity;
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
-import android.widget.Button;
+import android.view.View;
 import android.widget.EditText;
 
-public class DestinationAddActivity extends Activity {
+/**
+ * Activity that presents a user interface for entering information to 
+ * add a destination to a claim.
+ */
+public class DestinationAddActivity extends PlacePickerParentActivity {
+	//================================================================================
+	// Constants
+	//================================================================================
+	
+	/**
+	 * Intent key for the {@link Destination} object.
+	 */
+	public static final String EXTRA_DESTINATION = "com.indragie.cmput301as1.EXTRA_DESTINATION";
+	
+	/**
+	 * Intent key for the {@link User} object.
+	 */
+	public static final String EXTRA_USER = "com.indragie.cmput301as1.EXTRA_USER";
+	
+	/**
+	 * Intent key for checking if adding destination or editing.
+	 */
+	public static final String ADD_TO_CLAIM = "com.indragie.cmput301as1.ADD_TO_CLAIM";
+	
+	//================================================================================
+	// Properties
+	//================================================================================
+	
+	/**
+	 * Field for name of the destination.
+	 */
+	protected EditText nameField;
```

Comment:
Is there a reason for these not to be `private` instead? Is there a subclass of this class?